src/Shared/UnitTests/ObjectModelHelpers.cs

```diff
@@ -877,19 +877,10 @@ internal static void DeleteDirectory(string dir)
 
                     break;
                 }
-                catch (Exception ex)
+                // After all the retries fail, we fail with the actual problem instead of some difficult-to-understand issue later.
+                catch (Exception ex) when (retries < 4)
                 {
-                    if (retries < 4)
-                    {
-                        Console.WriteLine(ex.ToString());
-                    }
-                    else
-                    {
-                        // All the retries have failed. We will now fail with the
```

Comment:
Please consider to move this comment to new code somehow. I believe it is worthy comment.

Reply:
I put it right above the catch.